collect_app/src/main/java/org/odk/collect/android/activities/InstanceUploaderList.java

```diff
@@ -156,29 +162,39 @@ public void onUploadButtonsClicked(Button button) {
                     Context.CONNECTIVITY_SERVICE);
             NetworkInfo ni = connectivityManager.getActiveNetworkInfo();
 
-            if (NetworkReceiver.running) {
-                ToastUtils.showShortToast(R.string.send_in_progress);
-                return;
-            } else if (ni == null || !ni.isConnected()) {
+            if (ni == null || !ni.isConnected()) {
                 logger.logAction(this, "uploadButton", "noConnection");
                 ToastUtils.showShortToast(R.string.no_connection);
                 return;
             }
-        }
 
-        int checkedItemCount = getCheckedCount();
-        logger.logAction(this, "uploadButton", Integer.toString(checkedItemCount));
+            LiveData<List<WorkStatus>> statuses = WorkManager.getInstance().getStatusesForUniqueWork(AutoSendWorker.TAG);
 
-        if (checkedItemCount > 0) {
-            // items selected
-            uploadSelectedFiles(button.getId());
-            setAllToCheckedState(listView, false);
-            toggleButtonLabel(findViewById(R.id.toggle_button), listView);
-            uploadButton.setEnabled(false);
-            smsUploadButton.setEnabled(false);
-        } else {
-            // no items selected
-            ToastUtils.showLongToast(R.string.noselect_error);
+            statuses.observe(this, workStatuses -> {
+                if (workStatuses != null) {
+                    for (WorkStatus status : workStatuses) {
+                        if (status.getState().equals(State.RUNNING)) {
+                            ToastUtils.showShortToast(R.string.send_in_progress);
+                            return;
```

Comment:
This doesn't return out of `onUploadButtonsClicked`, does it? This means an upload can be triggered while an auto-send is ongoing which is not desired.

Reply:
Ah, or is this why the actual sending is done from within the observer?